### DISCUSSION TITLE
error: The verifier does not yet support the following Rust feature: complex tuple/struct patterns when trying to match using `matches!` macro

### DISCUSSION THREAD

**Discussion by tenzinhl** [verus-lang/verus#551] [General]
Hey there!

Clearly this is a known thing lacking support, but I just wanted to start this thread asking why it's currently unsupported and what level of work it would require to support?

I ran into the error `error: The verifier does not yet support the following Rust feature: complex tuple/struct patterns` when using the `matches!` macro to check that an enum instance is of a specific variant.

When I'm matching I don't care about the contents of the enum, so I tried writing:
```rust
matches!(step, CoordinationSystem::Step::commit_complete(..))
```

But that's when I got the error message "error: The verifier does not yet support the following Rust feature: complex tuple/struct patterns".

Currently to get around this I write:
```rust
matches!(step, CoordinationSystem::Step::commit_complete(_, _))
```

And that compiles fine. It seems like supporting `..` shouldn't be too bad, but is this error message is referring to more patterns
than just `..`?

**Comment by tenzinhl** [comment]
@utaal @tjhance (btw, do lemme know if I don't need to ping here)

**Comment by tjhance** [comment]
Yeah I expect this would be easy to support.

**Comment by tjhance** [comment]
This is supported as of bc267e31cbb189011c7ec12a83a5ed3aa68d4cac

**Reply by tenzinhl** [reply]
Thanks!
